components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/handlers/UsernameValidationConfigurationHandler.java

```diff
@@ -84,22 +85,20 @@ public ValidationConfiguration getDefaultValidationConfiguration(String tenantDo
             RealmConfiguration realmConfiguration = getRealmConfiguration(tenantDomain);
             String usernameRegEx = getUsernameRegEx(realmConfiguration);
 
-            // Return the JsRegex if the default regex has been updated by the user.
-            if (!usernameRegEx.isEmpty() &&
-                    !DEFAULT_EMAIL_JS_REGEX_PATTERN.equals(usernameRegEx)) {
-                if (isAlphaNumericValidationByDefault()) {
-                    rules.add(getDefaultLengthValidatorRuleConfig());
-                    rules.add(getRuleConfig(AlphanumericValidator.class.getSimpleName(),
-                            ENABLE_VALIDATOR, Boolean.TRUE.toString()));
-                    configuration.setRules(rules);
-                } else {
-                    rules.add(getRuleConfig("JsRegExValidator", JS_REGEX, usernameRegEx));
-                    configuration.setRegEx(rules);
-                }
-            } else {
-                rules.add(getRuleConfig(EmailFormatValidator.class.getSimpleName(),
+            if (!usernameRegEx.isEmpty() && !DEFAULT_EMAIL_JS_REGEX_PATTERN.equals(usernameRegEx)
+                    && isAlphaNumericValidationByDefault()) {
+                rules.add(getDefaultLengthValidatorRuleConfig());
+                rules.add(getRuleConfig(AlphanumericValidator.class.getSimpleName(),
                         ENABLE_VALIDATOR, Boolean.TRUE.toString()));
                 configuration.setRules(rules);
+            } else if (PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain().equals("carbon.super")) {
+                // Return the JSRegex if the tenant domain is carbon.super
```

Comment:
```suggestion
                // Return the JSRegex if the tenant domain is carbon.super.
```
And this preserves the existing behavior for Asgardeo users